File: pages/lib/model/Monitor.dart
import "package:pages/model/Schedule.dart";

class Monitor
{
  final String name;
  final String image;
  final Schedule schedule;

  Monitor(this.name, this.image, this.schedule);

  factory Monitor.fromJson(Map<String, dynamic> json) { return Monitor(json["name"], json["image"], Schedule.fromJson(json["schedule"])); }

  Map<String, dynamic> toJson()
  {
    return
    {
      "name": name,
      "image": image,
      "schedule": schedule.toJson()
    };
  }
}

File: api/model/Monitor.js
class Monitor {
    constructor(name, image, schedule) {
        this.name = name;
        this.image = image;
        this.schedule = schedule;
    }

    toJson() {
        return {
            name: this.name,
            image: this.image,
            schedule: this.schedule.toJson()
        };
    }
}

module.exports = { Monitor };